pydoctor/epydoc/markup/_types.py

```diff
@@ -5,64 +5,53 @@
 """
 from __future__ import annotations
 
-from typing import Any, Callable, Dict, List, Tuple, Union, cast
+from typing import Callable, Dict, List, Union
 
-from pydoctor.epydoc.markup import DocstringLinker, ParseError, ParsedDocstring, get_parser_by_name
-from pydoctor.node2stan import node2stan
+from pydoctor.epydoc.markup import ParsedDocstring
+from pydoctor.epydoc.markup._pyval_repr import PyvalColorizer
 from pydoctor.napoleon.docstring import TokenType, TypeDocstring
+from pydoctor.epydoc.docutils import new_document, set_node_attributes, code
 
 from docutils import nodes
-from twisted.web.template import Tag, tags
 
+# TODO: This class should use composition instead of multiple inheritence...
 class ParsedTypeDocstring(TypeDocstring, ParsedDocstring):
     """
     Add L{ParsedDocstring} interface on top of L{TypeDocstring} and 
-    allow to parse types from L{nodes.Node} objects, providing the C{--process-types} option.
+    allow to parse types from L{nodes.Node} objects, 
+    providing the C{--process-types} option.
     """
 
     FIELDS = ('type', 'rtype', 'ytype', 'returntype', 'yieldtype')
     
-    #                                                   yes this overrides the superclass type!
-    _tokens: list[tuple[str | nodes.Node, TokenType]] # type: ignore
-
-    def __init__(self, annotation: Union[nodes.document, str],
-                 warns_on_unknown_tokens: bool = False, lineno: int = 0) -> None:
+    def __init__(self, annotation: nodes.document, 
+                 warns_on_unknown_tokens: bool = False, 
+                 lineno: int = 0) -> None:
         ParsedDocstring.__init__(self, ())
-        if isinstance(annotation, nodes.document):
-            TypeDocstring.__init__(self, '', warns_on_unknown_tokens)
+        TypeDocstring.__init__(self, '', warns_on_unknown_tokens)
 
-            _tokens = self._tokenize_node_type_spec(annotation)
-            self._tokens = cast('list[tuple[str | nodes.Node, TokenType]]', 
-                                self._build_tokens(_tokens))
-            self._trigger_warnings()
-        else:
-            TypeDocstring.__init__(self, annotation, warns_on_unknown_tokens)
-        
+        _tokens = self._tokenize_node_type_spec(annotation)
+        # yes this overrides the superclass type!
```

Comment:
mypy is telling us that we override the superclass type but it would be more helpful to say why :)